requestmanager/requestmanager.go

```diff
@@ -485,12 +483,8 @@ func (prm *processResponseMessage) handle(rm *RequestManager) {
 	filteredResponses = rm.filterResponsesForPeer(filteredResponses, prm.p)
 	rm.updateLastResponses(filteredResponses)
 	responseMetadata := metadataForResponses(filteredResponses)
-	rm.asyncLoader.ProcessResponse(prm.p, responseMetadata, prm.blks)
+	rm.asyncLoader.ProcessResponse(responseMetadata, prm.blks)
 	rm.processTerminations(filteredResponses)
-	select {
-	case <-rm.ctx.Done():
-	case prm.response <- nil:
```

Comment:
Why isn't this still needed? Or was it not needed previously?

Reply:
prm.response as a field was removed in this PR I think. Waiting on the response channel was the essentially what was used to block on allocation, but that's now just being done by allocating directly in the receive handler.